Automattic/jetpack
# Slideshow Block: AMP Compatibility

#### Changes proposed in this Pull Request:
This PR makes the Slideshow block AMP compatible. This is achieved by using the [`amp-carousel`](https://amp.dev/documentation/components/amp-carousel) element instead of [Swiper](https://idangero.us/swiper/) in AMP requests. The functionality is similar although a few features are different and/or lost in the AMP version:

1) No support for Fade transition style in AMP version. Transitions will always be Slides. The `amp-carousel` slide transition is also faster and less graceful than Swiper's.
~2) There is no Autoplay pausing in the AMP version. If Autoplay is enabled it will always be on.~
~3) The pagination bullets don't show "active slide" state in the AMP version. They are functional (clicking will jump the slideshow to the requested slide) but they don't change state to indicate which is current.~
~4) Dynamic height sizing will not occur in the AMP version.~

Update: Autoplay pause/play button support added in https://github.com/Automattic/jetpack/pull/13009/commits/cabea0d9a74b9f3d6003d7df3224a78b966b6f10

Update: Pagination bullets active slide state in https://github.com/Automattic/jetpack/pull/13009/commits/64107c74dab5653099f48d952d7671d2590a2d88

Update: Dynamic height sizing in https://github.com/Automattic/jetpack/pull/13009/commits/3cfa9017b5e166a7c2db9a94426079e76f291fb6.

Part of https://github.com/Automattic/jetpack/issues/9730

#### Testing instructions:
* Spin up a Jurassic Ninja instance with this link: https://jurassic.ninja/create/?jetpack-beta&branch=add/amp-slideshow
* Set up Jetpack.
* Install and activate AMP: https://wordpress.org/plugins/amp/
* In AMP settings, set mode to *Transitional*
* Create a post, insert Slideshow block, add images.
* Publish and view the post, then switch to AMP by interacting with the AMP item in the admin menu, or by simply appending `?amp` to the URL.
* Observe that the slideshow looks and functions like the non-AMP version (taking into account the list of differences above).

#### Other testing scenarios
* Multiple slideshows on one page
* Slideshows with captions
* Captions with HTML tags

#### Proposed changelog entry for your changes:

* Slideshow block AMP compatibility
